Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/stolostron/builder:go1.22-linux AS builder
+FROM registry.ci.openshift.org/stolostron/builder:go1.23-linux AS builder
```

Comment:
follow-up:
- [ ] update the gitlab Dockerfile: https://gitlab.cee.redhat.com/mce-cpaas-midstream/multicloud-manager/-/blob/multicluster-engine-2.8-rhel-9/distgit/containers/multicloud-manager/Dockerfile.cached?ref_type=heads#L1